cmake/zlib.cmake

```diff
@@ -37,6 +37,8 @@ MACRO (MYSQL_CHECK_ZLIB_WITH_COMPRESS)
   IF(WITH_ZLIB STREQUAL "bundled")
     MYSQL_USE_BUNDLED_ZLIB()
   ELSE()
+    FIND_PACKAGE(PkgConfig QUIET)
```

Comment:
is, to depend on the finding of pkg-config is the following required:
```
IF(PKG_CONFIG_FOUND)
  PKG_GET_VARIABLE(ZLIB_ROOT zlib prefix)
ENDIF()
```

I'm roughly thinking it is based on https://gitlab.kitware.com/cmake/cmake/-/blob/master/Modules/FindPkgConfig.cmake#L131 .

Have you tried with pkg-config not there?

Reply:
> Have you tried with pkg-config not there?
Interesting, I moved `pkg-config` from `/usr/bin` to somewhere not in $PATH but the command can still find the right path:

```
-- Found ZLIB: /workspace/mylib/lib/libz.so (found version "1.2.13") 
```

I basically follow the convention here:
https://github.com/MariaDB/server/commit/3a33ae86010051a1022c392f2936b5fc9f3ab19e#diff-8e35291051929131abcdfd960c0e4e246317876c5ec2989154a8e810d489c7d6R81-R82, do you prefer me checking the availability here?

Reply:
I think its fine not that you've tested it. Thanks for the pcre reference implementation.